agent/agent.go

```diff
@@ -482,41 +464,36 @@ func (a *Agent) handleSetupError(ctx context.Context, tc *taskContext, err error
 	return tc, shouldExit, catcher.Resolve()
 }
 func shouldRunSetupGroup(nextTask *apimodels.NextTaskResponse, tc *taskContext) bool {
+	var previousTaskGroup string
+	if tc.taskConfig != nil && tc.taskConfig.TaskGroup != nil {
+		previousTaskGroup = tc.taskConfig.TaskGroup.Name
+	}
 	if !tc.ranSetupGroup { // we didn't run setup group yet
 		return true
 	} else if tc.taskConfig == nil ||
 		nextTask.TaskGroup == "" ||
 		nextTask.Build != tc.taskConfig.Task.BuildId { // next task has a standalone task or a new build
 		return true
-	} else if nextTask.TaskGroup != tc.taskGroup { // next task has a different task group
-		if tc.logger != nil && nextTask.TaskGroup == tc.taskConfig.Task.TaskGroup {
-			tc.logger.Task().Warning(message.Fields{
-				"message":                 "programmer error: task group in task context doesn't match task",
-				"task_config_task_group":  tc.taskConfig.Task.TaskGroup,
-				"task_context_task_group": tc.taskGroup,
-				"next_task_task_group":    nextTask.TaskGroup,
-			})
-		}
+	} else if nextTask.TaskGroup != previousTaskGroup { // next task has a different task group
 		return true
 	}
-
 	return false
 }
 
-func (a *Agent) fetchProjectConfig(ctx context.Context, tc *taskContext) error {
+func (a *Agent) fetchProjectConfig(ctx context.Context, tc *taskContext) (*task.Task, *model.Project, util.Expansions, map[string]bool, error) {
```

Comment:
this name seems inaccurate now (or maybe always was) that we don't add things to the task context. Maybe fetchTaskInfo?

Reply:
I renamed it. And yeah, it always fetched more stuff than just the project model.